cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcProvisioner.java

```diff
@@ -117,16 +117,17 @@ private String getMasterExternalIp(Cluster cluster) {
     return ip;
   }
 
-  // Name must start with a lowercase letter followed by up to 54 lowercase letters,
+  // Name must start with a lowercase letter followed by up to 51 lowercase letters,
   // numbers, or hyphens, and cannot end with a hyphen
   // We'll use app-runid, where app is truncated to fit, lowercased, and stripped of invalid characters
   @VisibleForTesting
   static String getClusterName(ProgramRun programRun) {
     String cleanedAppName = programRun.getApplication().replaceAll("[^A-Za-z0-9\\-]", "").toLowerCase();
-    int maxAppLength = 53 - programRun.getRun().length();
+    // 51 is max length, 5 is from 'cdap-' prefix, 1 is for '-' separating app name and run id
+    int maxAppLength = 51 - 5 - 1 - programRun.getRun().length();
```

Comment:
Better define `cdap-` as constant and then use `NAME_PREFIX.length()` instead of `- 5`